Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/ui/ViewBindingExtensions.kt

```diff
@@ -0,0 +1,57 @@
+package de.rki.coronawarnapp.util.ui
+
+import android.os.Handler
+import android.os.Looper
+import android.view.View
+import androidx.annotation.MainThread
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.viewbinding.ViewBinding
+import kotlin.properties.ReadOnlyProperty
+import kotlin.reflect.KProperty
+
+inline fun <FragmentT : Fragment, reified BindingT : ViewBinding> FragmentT.viewBindingLazy() =
+    viewBindingLazy {
+        val bindingMethod = BindingT::class.java.getMethod("bind", View::class.java)
+        bindingMethod(null, requireView()) as BindingT
+    }
+
+@Suppress("unused")
+fun <FragmentT : Fragment, BindingT : ViewBinding> FragmentT.viewBindingLazy(
+    bindingProvider: (FragmentT) -> BindingT
+) = ViewBindingProperty(bindingProvider)
+
+class ViewBindingProperty<ComponentT : LifecycleOwner, BindingT : ViewBinding>(
+    private val bindingProvider: (ComponentT) -> BindingT
+) : ReadOnlyProperty<ComponentT, BindingT> {
+
+    private val uiHandler = Handler(Looper.getMainLooper())
+    private var localRef: ComponentT? = null
+    private var viewBinding: BindingT? = null
+
+    private val onDestroyObserver = object : DefaultLifecycleObserver {
+        // Called right before Fragment.onDestroyView
+        override fun onDestroy(owner: LifecycleOwner) {
+            val ref = localRef ?: return
```

Comment:
In this case we would loose more readability due to the double nesting, imho.

```kotlin
        override fun onDestroy(owner: LifecycleOwner) {
            localRef?.let {
                it.lifecycle.removeObserver(this)
                localRef = null
                // Otherwise the binding is null before Fragment.onDestroyView
                uiHandler.post { viewBinding = null }
            }
        }
```

vs

```kotlin
        override fun onDestroy(owner: LifecycleOwner) {
            val ref = localRef ?: return
            ref.lifecycle.removeObserver(this)
            localRef = null
            // Otherwise the binding is null before Fragment.onDestroyView
            uiHandler.post { viewBinding = null }
        }
```

Unless you feel strongl about this, i'll leave it as is.